File: day1.py
def calc(words):
    thesum = 0
    spell = ["", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]
    vals = {}
    for i in range(1, 10):
        if words: vals[spell[i]] = i
        vals[str(i)] = i
        
    for line in open("day1.txt").read().split("\n"):
        hits = []
        for x in range(len(line)):
            for string in vals:
                if line[x:].startswith(string):
                    hits.append(vals[string])
        thesum += hits[0]*10 + hits[-1]
    return thesum
    
print(calc(False))
print(calc(True))

File: Day1.java
//Advent of Code Day 1

import java.io.*;
import java.util.*;

public class HelloWorld {
	public static void main(String[] args) throws FileNotFoundException {
		System.out.println(calc(false));
		System.out.println(calc(true));
	}
	public static int calc(boolean words) throws FileNotFoundException {
		ArrayList<String> filelist = new ArrayList<String>();
		Scanner s = new Scanner(new File("day1.txt"));
		while (s.hasNext()){
		    filelist.add(s.next());
		}
		s.close();
		Integer thesum = 0;
		ArrayList<Integer> hits = new ArrayList<Integer>();
		HashMap<String, Integer> vals = new HashMap<String, Integer>();
		String[] nums = {"", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
		
		for(int x = 1; x < 10; x++) {
			if(words) vals.put(nums[x], x);
			vals.put(Integer.toString(x), x);
		}
	
		for(String line: filelist) {
			hits.clear();
			for(int x = 0; x < line.length(); x++) {
				for(String str: vals.keySet()) {
					if(line.substring(x,line.length()).startsWith(str)){
						hits.add(vals.get(str));
					}
				}
			}
			thesum += hits.get(0)*10 + hits.get(hits.size()-1);
		}
		return thesum;
	}
}
